src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java

```diff
@@ -71,7 +77,10 @@ public PatronNoticeService(PatronNoticePolicyRepository noticePolicyRepository,
 
   private CompletableFuture<Result<Pair<PatronNoticeEvent, String>>> loadNoticePolicyId(PatronNoticeEvent event) {
     return noticePolicyRepository.lookupPolicyId(event.getItem(), event.getUser())
-      .thenApply(mapResult(circulationRuleMatchEntity -> Pair.of(event, circulationRuleMatchEntity.getPolicyId())));
+      .thenApply(mapResult(circulationRuleMatchEntity -> {
+        event.getNoticeContext().put(NOTICE_POLICY_ID, circulationRuleMatchEntity.getPolicyId());
```

Comment:
Can this side effect will be done after receiving response from the method?

Alternatively,  we can reflect the side effect in method name, e.g. - `loadNoticePolicyIdAndPutInContext`